### PR TITLE
Feat/15704056 filter pills

### PR DESCRIPTION
**Pull request by CzarekDryl**
[tw task](https://tw.auroracreation.com/app/tasks/15704056)

## Description

- Added filters to funding page

Acceptance criteria

- user should see a pills for Active Filters
- user should be able to close pills and by that reset filter

### CODE REVIEW

**Comment by CzarekDryl** [comment]
@iamsamgibbs I've missed one place when renaming variables 😢

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
Nice job! I only noticed 1 thing.

I unmarked the `Custom` filter, yet the "subfilters" were still active:
Screenshot 2024-02-28 at 13 11 12

**Comment by CzarekDryl** [comment]
@ArmandoGraterol I think you've checked wrong page. Filters that I added are on `/incoming` page.

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Good work on this so far, almost ready to go.

I feel like the 'Token status' filter should work similarly to the 'User status' filter on the members page where it can be either option or off to show both. Currently when neither option is selected you get no results at all.

Screenshot 2024-02-29 at 08 46 16

This also leads to inconsistent behaviour as 'approved' can be deselected from the menu, but the filter tag cannot be cleared.

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Thanks for your work @CzarekDryl, I'm also noticing the issue Sam raised.

Also, the pill for status filter doesn't appear initially, but after you change it to something else and back to the default value, the pill appears but it's impossible to remove it:

https://github.com/JoinColony/colonyCDapp/assets/112586815/2e6e4333-175b-48fd-be42-742b6bd943b5

File: src/components/frame/v5/pages/FundsPage/partials/FundsTable/FundsTable.tsx [hunk 1]
```diff
@@ -21,16 +23,95 @@ const displayName = 'pages.FundsPage.partials.FundsTable';
...
     new Set(claims.map((claim) => claim.token?.tokenAddress || '')),
   );
 
+  const activeFilters = filters.items
+    .map((item) => {
+      const activeItem = filters.value[item.name];
+
+      if (item.name === 'status' && !isStatusChanged) {
+        return undefined;
+      }
```
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/frame/v5/pages/FundsPage/partials/FundsTable/FundsTable.tsx]: At the beginning, pills with the default value should not be shown, only after any change do we show the pills. So I think that I need this state to check that.

image

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Thanks for the improvements, it's very nearly there!

I've noticed this issue with the filters. `ƓƓƓ` is approved, so it shouldn't show with this selection of filters.

Screenshot 2024-03-04 at 09 14 48

**Review by ArmandoGraterol — COMMENTED** [review]
Sorry! I was indeed looking at the wrong page.

- If I apply both status filters, only the approved is shown, is this intentional?
Screenshot 2024-03-04 at 13 27 49

**Comment by CzarekDryl** [comment]
@iamsamgibbs @ArmandoGraterol Filtering should work as expected now, please check it again :)

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Good work! Filters are working great now. I just noticed on the figma design that "Token type" should appear before "Token status" in the menu, if you could swap those over then we're all good to go!

Screenshot 2024-03-05 at 14 13 56

**Review by ArmandoGraterol — DISMISSED** [review]
Looking good to me 👍 Thank you for the changes

**Review by jakubcolony — DISMISSED** [review]
Looks like all the previous issues are now resolved, nice work @CzarekDryl!

**Review by warrencolliver — APPROVED** [review]
Nice work on this. tested the changes and I can't find any problems

**Pull request merged**